crypto/stateproof/structs.go

```diff
@@ -82,6 +82,10 @@ type StateProof struct {
 	PositionsToReveal []uint64          `codec:"pr,allocbound=MaxReveals"`
 }
 
+// SigPartProofMaxSize is the maximum valid size of SigProofs and PartProofs elements of the Stateproof struct in bytes.
+// It is equal to merklearray.ProofMaxSizeByElements(config.StateProofTopVoters/2)
```

Comment:
What is the reason for having this `config.StateProofTopVoters/2` and not `config.StateProofTopVoters`?

Reply:
look at `merklearray.Proof` definition and a comment there

>	// Path is bounded by MaxNumLeavesOnEncodedTree since there could be multiple reveals, and
	// given the distribution of the elt positions and the depth of the tree,
	// the path length can increase up to 2^MaxEncodedTreeDepth / 2
	Path        []crypto.GenericDigest `codec:"pth,allocbound=MaxNumLeavesOnEncodedTree/2"`